### PR TITLE
[feat] - concurently scan the filesystem source

### PR DESCRIPTION
**Pull request by ahrav**
### Description:
concurrently scan files within the fileystem source.

### Checklist:
* [ ] Tests passing (`make test-community`)?
* [ ] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/usage/install/#local-installation))?

### CODE REVIEW

File: pkg/sources/filesystem/filesystem.go [hunk 1]
```diff
@@ -93,7 +96,7 @@ func (s *Source) Chunks(ctx context.Context, chunksChan chan *sources.Chunk, _ .
 		fileInfo, err := os.Stat(cleanPath)
 		if err != nil {
 			logger.Error(err, "unable to get file info")
-			continue
+			return nil
```
Comment (rosecodym) [inline pkg/sources/filesystem/filesystem.go:99]: This seems like a really big change! Is it related to concurrency? Should it be in this PR?

File: pkg/sources/filesystem/filesystem_test.go [hunk 2]
```diff
@@ -71,16 +72,23 @@ func TestSource_Scan(t *testing.T) {
...
 					return
 				}
 			}()
-			gotChunk := <-chunksCh
-			if diff := pretty.Compare(gotChunk.SourceMetadata, tt.wantSourceMetadata); diff != "" {
-				t.Errorf("Source.Chunks() %s diff: (-got +want)\n%s", tt.name, diff)
+			var counter int
+			for chunk := range chunksCh {
+				counter++
+				if chunk.SourceMetadata.GetFilesystem().GetFile() == "filesystem.go" {
```
Comment (rosecodym) [inline pkg/sources/filesystem/filesystem_test.go:85]: Do you think it's worth checking that this ever happened?

**Review by rosecodym — COMMENTED** [review]
The concurrency looks as straightforward as concurrency can ever be (not much) but I'm nervous about the added early return elsewhere.

File: pkg/sources/filesystem/filesystem.go (hunk 1 above)
Comment (ahrav) [reply pkg/sources/filesystem/filesystem.go]: Whoops this was left behind when we had first put the concurrency within this level. reverted it.

File: pkg/sources/filesystem/filesystem.go [hunk 3]
```diff
@@ -126,9 +135,13 @@ func (s *Source) scanDir(ctx context.Context, path string, chunksChan chan *sour
 			return nil
 		}
 
-		if err = s.scanFile(ctx, fullPath, chunksChan); err != nil {
-			ctx.Logger().Info("error scanning file", "path", fullPath, "error", err)
-		}
+		workerPool.Go(func() error {
+			if err = s.scanFile(ctx, fullPath, chunksChan); err != nil {
+				ctx.Logger().Info("error scanning file", "path", fullPath, "error", err)
```
Comment (mcastorina) [inline pkg/sources/filesystem/filesystem.go:140]: Should this be an `Error` log instead of `Info`?

Comment (ahrav) [reply pkg/sources/filesystem/filesystem.go:140]: yea probably.

File: pkg/sources/filesystem/filesystem_test.go (hunk 2 above)
Comment (ahrav) [reply pkg/sources/filesystem/filesystem_test.go:84]: I can use the counter to capture this happening.

**Review by mcastorina — APPROVED** [review]

**Pull request merged**